ops/framework.py

```diff
@@ -124,14 +124,14 @@ def __init__(self, parent: Optional[Union['Handle', 'Object']], kind: str, key:
         self._key = key
         if parent:
             if key:
-                self._path = "{}/{}[{}]".format(parent, kind, key)
+                self._path = f"{parent}/{kind}[{key}]"
             else:
-                self._path = "{}/{}".format(parent, kind)
+                self._path = f"{parent}/{kind}"
         else:
             if key:
-                self._path = "{}[{}]".format(kind, key)
+                self._path = f"{kind}[{key}]"
             else:
-                self._path = "{}".format(kind)
+                self._path = kind
```

Comment:
what if kind isn't a string?

Reply:
Yeah, it should always have been, but this is Python so you never know. :-) Anyway, I reverted back to `f"{kind}"` mainly because it's consistent with the above so less surprising (and the str thing), and added a comment as to why.